test/src/main/java/org/bf2/test/TestUtils.java

```diff
@@ -97,15 +97,15 @@ public void run() {
         return future;
     }
 
-    public static Path getLogPath(String folderName, ExtensionContext context) {
+    public static Path getLogPath(String folderName, ExtensionContext context) throws NullPointerException {
         String testMethod = context.getDisplayName();
-        Class<?> testClass = context.getTestClass().get();
+        Class<?> testClass = context.getTestClass().orElse(null);
         return getLogPath(folderName, testClass, testMethod);
     }
 
-    public static Path getLogPath(String folderName, TestInfo info) {
+    public static Path getLogPath(String folderName, TestInfo info) throws NullPointerException {
         String testMethod = info.getDisplayName();
-        Class<?> testClass = info.getTestClass().get();
+        Class<?> testClass = info.getTestClass().orElse(null);
         return getLogPath(folderName, testClass, testMethod);
     }
```

Comment:
The `orElse` paths here are guaranteed NPEs. Let's think about an alternate approach. Perhaps have the `getLogPath` called by these two methods to accept a `String` rather than the class name, then change these to map the class name orElse a constant "NOCLASS". For example:/
```java
String testClassName = info.getTestClass().map(Class::getName).orElse("NOCLASS");
```

Reply:
Right! But shouldn't we `log` it when it hits `NOCLASS`?